design/lvm/resize_workflow.md

```diff
@@ -0,0 +1,151 @@
+---
+title: LVM-LocalPV Volume Expansion
+authors:
+  - "@pawanpraka1"
+owners:
+  - "@kmova"
+creation-date: 2021-05-28
+last-updated: 2021-05-28
+status: Implemented
+---
+
+# LVM-LocalPV Volume Expansion
+
+## Table of Contents
+
+- [LVM-LocalPV Volume Expansion](#lvm-localpv-volume-expansion)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+    - [Implementation Details](#implementation-details)
+        - [Controller Expansion](#controller-expansion)
+        - [Filesystem Expansion](#filesystem-expansion)
+    - [Steps to perform volume expansion](#steps-to-perform-volume-expansion)
+    - [High level Sequence Diagram](#high-level-sequence-diagram)
+    - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+
+## Summary
+
+This proposal charts out the design details to implement expansion workflow on LVM-LocalPV Volumes.
+
+## Motivation
+
+### Goals
+
+- As a user, I should be able to resize the volumes provisioned via LVM-LocalPV by updating the size
+  on PersistentVolumeClaim(PVC).
+
+### Non-Goals
+
+- Support for volume expansion when underlying volume has lvm snapshots.
+
+## Proposal
+
+### User Stories
+
+As an application developer, I should be able to resize volume on the fly(When application consuming volume).
+
+Steps To Be Performed By User:
+1. Update the PVC capacity using `kubectl edit pvc <pvc_name>`
+
+Note: StorageClass should allow expanding the volume(i.e allowVolumeExpansion field must set to true).
+
+### Implementation Details
+
+LVM volume expansion is a two-step process(controller expansion & filesystem expansion) and
+it gets triggered upon updating PVC capacity.
+
+##### Controller Expansion
+
+- External-resizer is a sidecar in `openebs-lvm-controller` pod which watches for PVC capacity
+  updates. Once it receives the event external-resizer will make a `ControllerExpandVolume` gRPC request
+  to LVM-CSI controller plugin.
+- LVM-CSI controller plugin is another container in the same pod where external-resizer is running.
+  Once the controller plugin receives the `ControllerExpandVolume` gRPC request it does the following steps:
+  - List active snapshots on resizing volume, if there are any snapshots plugin will return an error
```

Comment:
can we explain why we are returning error here?

Reply:
Updated the doc